### PR TITLE
EG-1169, Added FormPage, Split viewer errors into Required and Type

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/controllers/GuidanceController.scala [hunk 1]
```diff
@@ -54,9 +52,19 @@ class GuidanceController @Inject() (
...
     withExistingSession[PageContext](service.getPageContext(processCode, s"/$path", p.isDefined, _)).flatMap {
-      case Right(pageContext) =>
-        showPage(path, pageContext)
+      case Right(pageCtx) =>
+        logger.info(s"Retrieved page at ${pageCtx.page.urlPath}, start at ${pageCtx.processStartUrl}," +
+                    s" answer = ${pageCtx.answer}, backLink = ${pageCtx.backLink}")
+        pageCtx.page match {
+          case page: StandardPage => service.saveLabels(pageCtx.sessionId, pageCtx.labels).map {
+              case Right(_) => Ok(standardView(page, pageCtx))
+              case Left(err) => InternalServerError(errorHandler.internalServerErrorTemplate)
```
Comment (LesWilson) [inline app/controllers/GuidanceController.scala:63]: err not used

File: app/services/GuidanceService.scala [hunk 2]
```diff
@@ -59,21 +58,27 @@ class GuidanceService @Inject() (
               logger.error(s"PageBuilder error $err on process ${process.meta.id} with sessionId $sessionId")
               Left(InvalidProcessError)
             },
-            page =>
+            page => {
+              val stanzaIdToUrlMap = urlToPageId.map{case (k, v) => (v, s"${appConfig.baseUrl}/${processCode}${k}")}.toMap
```
Comment (LesWilson) [inline app/services/GuidanceService.scala:62]: don't need toMap

File: app/controllers/GuidanceController.scala (hunk 1 above)
Comment (cmrhmrc) [reply app/controllers/GuidanceController.scala]: replace with _

File: app/services/GuidanceService.scala (hunk 2 above)
Comment (cmrhmrc) [reply app/services/GuidanceService.scala]: removed

**Review by LesWilson — APPROVED** [review]
Looks good :)

**Pull request merged**